libgadget/sfr_eff.c

```diff
@@ -66,56 +66,62 @@ void cooling_and_starformation(void)
     const int nactive = NumActiveParticle;
 
     if(All.StarformationOn) {
-        NewStars = mymalloc("NewStars", nactive * sizeof(int) * All.NumThreads);
-        gadget_setup_thread_arrays(NewStars, thrqueuesfr, nqthrsfr, nactive, All.NumThreads);
-        NewParents = mymalloc2("NewParents", nactive * sizeof(int) * All.NumThreads);
-        gadget_setup_thread_arrays(NewParents, thrqueueparent, nqthrsfr, nactive, All.NumThreads);
+        int narr = ceil(nactive/All.NumThreads);
```

Comment:
I would avoid floating point ops here. You can rely on integer divide and add 1 to make sure it is large enough.

Reply:
Indeed! The current code in fact has an off-by-one error.